checker/manual/creating-a-checker.tex

```diff
@@ -1094,31 +1087,84 @@ \section{The checker class:  Compiler interface\label{writing-compiler-interface
 checker classes for the Interning Checker and the Nullness Checker:
 
 \begin{Verbatim}
-  @TypeQualifiers({ Interned.class, PolyInterned.class, PolyAll.class })
   @SupportedLintOptions({"dotequals"})
   public final class InterningChecker extends BaseTypeChecker { }
 
-  @TypeQualifiers({ Nullable.class, Raw.class, NonNull.class, PolyNull.class, PolyAll.class })
   @SupportedLintOptions({"flow", "cast", "cast:redundant"})
   public class NullnessChecker extends BaseTypeChecker { }
 \end{Verbatim}
 
-
 The checker class must indicate the annotations
 that make up the type hierarchy for this checker (including
-polymorphic qualifiers), either via a
-\refqualclass{framework/qual}{TypeQualifiers} annotation or by overriding
+polymorphic qualifiers), either by placing the annotations inside a subfolder called \<qual> located in the same directory as the checker, or by overriding
 the
 \refmethodterse{framework/type}{AnnotatedTypeFactory}{createSupportedTypeQualifiers}{--}
-method.  Each argument to \<@TypeQualifiers> or value returned by
-\<createSupportedTypeQualifiers> is a class literal for a type qualifier
-whose definition bears the
-\refqualclass{framework/qual}{TypeQualifier} meta-annotation.
+method. By default, the polymorphic qualifier \refqualclass{framework/qual}{PolyAll} is supported by a checker. It can be optionally excluded.
+
+There are four recommended ways to write an override implementation of \refmethodterse{framework/type}{AnnotatedTypeFactory}{createSupportedTypeQualifiers}{--}
+
+\begin{enumerate}
+\item using reflective lookup to add all qualifiers, and exclude \code{@PolyAll}. Example:
+
+\begin{Verbatim}
+  @Override
+  protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+      return loadTypeQualifiersFromQualDir(true, null);
+  }
+\end{Verbatim}
+
+\item using reflective lookup to add qualifiers, exclude \code{@PolyAll}, and add a manual
+list of qualifiers. Example:
+
+\begin{Verbatim}
+@Override
+protected Set<Class<? extends Annotation>> createSupportedTypeQualifiers() {
+    return loadTypeQualifiersFromQualDir(true,
+        new HashSet<Class<? extends Annotation>>(
+            Arrays.asList(H1Top.class, H1S1.class, H1S2.class,
+                H1Bot.class, H2Top.class, H2S1.class, H2S2.class,
+                H2Bot.class, H1Poly.class, H2Poly.class)));
+}
+\end{Verbatim}
+
+The use of a manual list is necessary for any qualifiers that are not located in
+the \<qual> folder, or for qualifiers which do not or cannot have the
```

Comment:
Change "which" to "that".

I'm not trying to be pedantic, just to make the text read better.  FYI, here is the rule:

That.  Which.  _That_ is the defining, or restrictive pronoun, _which_ the
nondefining, or nonrestrictive.
	The lawn mower that is broken is in the garage.  (Tells which one.)
	The lawn mover, which is broken, is in the garage.  (Adds a fact
		about the only mower in question.)
The use of _which_ for _that_ is common in written and spoken language
("Let us now go even unto Bethlehem, and see this thing which is come to
pass.")  Occasionally _which_ seems preferable to _that_, as in the
sentence from the Bible.  But it would be a convenience to all if these two
pronouns were used with precision.  The careful writer, watchful for small
conveniences, goes _which_-hunting, removes the defining _whiches_, and by
so doing improves his work.
				The Elements of Style
				William Strunk, Jr. and E. B. White